torch/lib/THD/base/data_channels/DataChannelTCP.hpp

```diff
@@ -87,6 +87,8 @@ struct DataChannelTCP : DataChannel {
   std::vector<Process> _processes; // Other processes in network
   std::unique_ptr<struct pollfd[]> _poll_events; // Events array for `poll`
 
+  std::mutex _mutex; // General mutex for methods - to make methods run atomically.
```

Comment:
I think a better description would be to say that it's there `to protect access to the TCP channel`